sv_pipeline/genome/load_data.py

```diff
@@ -0,0 +1,210 @@
+import argparse
+import hail as hl
+import logging
+import os
+import time
+
+from hail_scripts.v02.utils.elasticsearch_client import ElasticsearchClient
+
+from sv_pipeline.utils.common import get_sample_subset, get_sample_remap, get_es_index_name, CHROM_TO_XPOS_OFFSET
+from sv_pipeline.genome.utils.mapping_gene_ids import load_gencode
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+EXP_CHROM_TO_XPOS_OFFSET = hl.literal(CHROM_TO_XPOS_OFFSET)
+
+WGS_SAMPLE_TYPE = 'WGS'
+
+INTERVAL_TYPE = 'array<struct{type: str, chrom: str, start: int32, end: int32}>'
+
+BASIC_FIELDS = {
+    'contig': lambda rows: rows.locus.contig.replace('^chr', ''),
+    'sc': lambda rows: rows.info.AC[0],
+    'sf': lambda rows: rows.info.AF[0],
+    'sn': lambda rows: rows.info.AN,
+    'start': lambda rows: rows.locus.position,
+    'end': lambda rows: hl.if_else(hl.is_defined(rows.info.END2), rows.info.END2, rows.info.END),
+    'sv_callset_Het': lambda rows: rows.info.N_HET,
+    'sv_callset_Hom': lambda rows: rows.info.N_HOMALT,
+    'gnomad_svs_ID': lambda rows: rows.info.gnomAD_V2_SVID,
+    'gnomad_svs_AF': lambda rows: rows.info.gnomAD_V2_AF,
+    'pos': lambda rows: rows.locus.position,
+    'filters': lambda rows: hl.array(rows.filters.filter(lambda x: x != 'PASS')),
+    'xpos': lambda rows: get_xpos(rows.locus.contig, rows.locus.position),
+    'cpx_intervals': lambda rows: hl.if_else(hl.is_defined(rows.info.CPX_INTERVALS),
+                                             rows.info.CPX_INTERVALS.map(lambda x: get_cpx_interval(x)),
+                                             hl.null(hl.dtype(INTERVAL_TYPE))),
+}
+
+COMPUTED_FIELDS = {
+    'xstart': lambda rows: rows.xpos,
+    'xstop': lambda rows: hl.if_else(hl.is_defined(rows.info.END2),
+                                     get_xpos(rows.info.CHR2, rows.info.END2),
+                                     get_xpos(rows.locus.contig, rows.info.END)),
+    'svType': lambda rows: rows.sv_type[0],
+    'transcriptConsequenceTerms': lambda rows: [rows.sv_type[0]],
+    'sv_type_detail': lambda rows: hl.if_else(rows.sv_type[0] == 'CPX', rows.info.CPX_TYPE,
+                                              hl.if_else(rows.sv_type[0] == 'INS',
+                                                         rows.sv_type[-1], hl.null('str'))),
+    'sortedTranscriptConsequences': lambda rows: rows.gene_affected,
```

Comment:
really cool that `gene_affected` doesn't need any additional processing here! It might be neater if you annotate `sortedTranscriptConsequences` directly and skip the `gene_affected` list altogether

Reply:
Cool.

Reply:
I won't add `sortedTranscriptConsequences` to the `BASIC_FIELDS` but just change `gene_affected` to `sortedTranscriptConsequences` in the annotate_fields method. It is because the annotation needs the `gene_id_mapping` which is accessible with the method.